blog/_posts/2024-12-04-mirrord-on-aks.md

```diff
@@ -0,0 +1,184 @@
+
+You'll need:
+- An AKS cluster with the store demo deployed (you’ll need the full deployment from  `aks-store-all-in-one.yaml` for this one), accessible from your machine with `kubectl`
+- Go (available from [their website](https://go.dev/doc/install))
+- A local clone of the store demo
+
+Open `makeline-service` in VSCode and make sure mirrord is enabled (refer to the [Rust example](A practical example: Rust) for instructions). Next, open mirrord’s configuration by hovering over the `mirrord` status icon in the bottom bar and clicking “Settings”. Replace the contents of the `mirrord.json` file that was created with the following:
+
+```json
+{
+	"feature": {
+    	"network": {
+        	"incoming": "steal",
+        	"outgoing": true
+    	},
+    	"fs": "read",
+    	"env": true
+	},
+	"target": "deployment/makeline-service"
+}
+```
+
+Keep the `store-admin` page open so that we can track orders (grab the IP with `kubectl get service store-admin --output 'jsonpath={..status.loadBalancer.ingress[0].ip}'`) and make sure that new orders are being placed:
+
+![Screenshot of the pet store admin page](/AKS/assets/images/mirrord-on-aks/aks-pet-store-admin-page.png)
+
+Say, for example, that you have had a genius new idea regarding order spam detection: you want to try discarding orders in the queue that contain more than one type of item.
+
+Clicking on some of the orders, you can see that they’re a random collection of usually a few different types of item, like this (note that orders are created in a way such that some will have multiple entries for the same kind of item - we think these are spam too, and we only want to process orders with a single item entry):
+
+![Screenshot of the pet store orders](/AKS/assets/images/mirrord-on-aks/aks-pet-store-orders.png)
+
+We don’t want these, they’re (possibly) spam! Change the contents of `main.go` as follows: replace lines 77-78 with this code for our new anti-spam measure:
+
+```go
+// NEW: Discard orders containing more than 1 different type of item
+var new_orders []Order
+for _, order := range orders {
+	if len(order.Items) <= 1 {
+    	new_orders = append(new_orders, order)
+	}
+}
+
+// Save orders to database
+err = client.repo.InsertOrders(new_orders)
+```
+
+Re-run the debugger, refresh the `store-admin` page and check for new orders - they should all be orders with only one type of item, which you can check by clicking on the order number:
+
+![Screenshot of the pet store admin orders](/AKS/assets/images/mirrord-on-aks/aks-pet-store-admin-orders.png)
+
+Much better. This means that mirrord not only stole traffic for the `makeline-service` deployment - it also read messages from RabbitMQ and wrote new entries to MongoDB, all without messing around with configuration and credentials! It can do this because by targeting the `makeline-service` pod, it reads the environment variables and files from that pod in the cluster.
```

Comment:
I found the word "stole" a bit inaccurate and disturbing, so replicate might be a better word, or maybe consult GPT

Reply:
Stole is the correct word here, they use it in their official documentation: https://mirrord.dev/docs/using-mirrord/steal/